### ISSUE TITLE
Always get Layer 'XX' already exists, please choice other 'name' or reuse this layer

### ISSUE THREAD

**Issue by Nandayang** [zsdonghao/tensorlayer#146]
i have already set reuse,but it still happens when i run the code.why?plz help me! thanks a lot!!!!! i always get 
:\nHint : Use different name for different 'Layer' (The name is used to control parameter sharing)" % name)

Exception: Layer 'model/input' already exists, please choice other 'name' or reuse this layer
Hint : Use different name for different 'Layer' (The name is used to control parameter sharing) i use tensorflow1.1.  <br>
so whats the paoblem?
```
import tensorflow as tf 
import tensorlayer as tl
from tensorlayer.layers import set_keep
import numpy as np
import time

X_train,y_train,X_test,y_test = tl.files.load_cifar10_dataset()
X_val = X_train[45000:50000,]
y_val = y_train[45000:50000,]
X_train = X_train[:45000,]
y_train = y_train[:45000,]



def model(x,y_,is_train,reuse):
    tl.layers.set_name_reuse(reuse)
    W_init = tf.contrib.layers.xavier_initializer()
    W_init2 = tf.truncated_normal_initializer(stddev=0.04)
    b_init2 = tf.constant_initializer(value=0.1)
    with tf.variable_scope("model1", reuse=reuse):
        network = tl.layers.InputLayer(x,name='input')
        network = tl.layers.Conv2dLayer(network,
                        act = tf.nn.relu,
                        shape = [5, 5, 3, 192],  # 64 features for each 5x5x3 patch
                        strides=[1, 1, 1, 1],
                        padding='SAME',
                        W_init=W_init, b_init=None,
                        name ='cnn_layer1')     # output: (?, 32, 32, 64)
        network = tl.layers.BatchNormLayer(network,is_train,name='batch_1')
        network = tl.layers.PoolLayer(network,
                        ksize=[1, 3, 3, 1],
                        strides=[1, 2, 2, 1],
                        padding='SAME',
                        pool = tf.nn.max_pool,
                        name ='pool_layer1')   # output: (?, 16, 16, 64)

        network = tl.layers.DropoutLayer(network, keep=0.6, name='drop1')
        network = tl.layers.Conv2dLayer(network,
                        act = tf.nn.relu,
                        shape = [3, 3, 192, 192], # 64 features for each 5x5 patch
                        strides=[1, 1, 1, 1],
                        padding='SAME',
                        W_init=W_init, b_init=None,
                        name ='cnn_layer2')     # output: (?, 16, 16, 64)
        network = tl.layers.BatchNormLayer(network,is_train,name='batch_2')


        network = tl.layers.PoolLayer(network,
                        ksize=[1, 3, 3, 1],
                        strides=[1, 2, 2, 1],
                        padding='SAME',
                        pool = tf.nn.max_pool,
                        name ='pool_layer2')   # output: (?, 8, 8, 64)

        network = tl.layers.Conv2dLayer(network,
                        act = tf.nn.relu,
                        shape = [3, 3, 192, 96],  # 64 features for each 5x5x3 patch
                        strides=[1, 1, 1, 1],
                        padding='SAME',
                        W_init=W_init, b_init=None,
                        name ='cnn_layer3')   

        network = tl.layers.BatchNormLayer(network,is_train,name='batch_3')

        network = tl.layers.PoolLayer(network,
                        ksize=[1, 3, 3, 1],
                        strides=[1, 2, 2, 1],
                        padding='SAME',
                        pool = tf.nn.max_pool,
                        name ='pool_layer3') 

        network = tl.layers.Conv2dLayer(network,
                        act = tf.nn.relu,
                        shape = [3, 3, 96, 96],  # 64 features for each 5x5x3 patch
                        strides=[1, 1, 1, 1],
                        padding='SAME',
                        W_init=W_init, b_init=None,
                        name ='cnn_layer4')

        network = tl.layers.BatchNormLayer(network,is_train,name='batch_4')

        network = tl.layers.PoolLayer(network,
                        ksize=[1, 3, 3, 1],
                        strides=[1, 2, 2, 1],
                        padding='SAME',
                        pool = tf.nn.max_pool,
                        name ='pool_layer4') 



        network = tl.layers.FlattenLayer(network, name='flatten_layer')
                                                                # output: (?, 4096)
        network = tl.layers.DenseLayer(network, n_units=1024,act = tf.nn.relu,
                                   W_init=W_init2, b_init=b_init2, name='relu1') # output: (?, 384)
        network = tl.layers.DenseLayer(network, n_units=382,
                            act = tf.nn.relu, W_init=W_init2, b_init=b_init2, name='relu2') # output: (?, 192)
        network = tl.layers.DenseLayer(network, n_units=10,
                            act = tf.identity,W_init=tf.truncated_normal_initializer(stddev=1/192.0),
                            name='output_layer')            # output: (?, 10)
        y = network.outputs
    
        cost = tl.cost.cross_entropy(y,y_,name='softmaxloss')
        L2 = tf.contrib.layers.l2_regularizer(0.004)(network.all_params[8]) + \
                    tf.contrib.layers.l2_regularizer(0.004)(network.all_params[10])
        loss = cost + L2            
        correct_pred = tf.equal(tf.argmax(y,1),y_)
        acc = tf.reduce_mean(tf.cast(correct_pred,tf.float32))
    
        return network,loss,acc

tf.reset_default_graph()
sess = tf.InteractiveSession()
tl.layers.set_name_reuse(True)

x = tf.placeholder(tf.float32, shape=[None, 32, 32, 3], name='x')
y_ = tf.placeholder(tf.int64, shape=[None, ], name='y_')


network, cost, _ = model(x, y_,is_train=True,reuse=True)
_, cost_test, acc = model(x, y_, is_train=False,reuse=True)

n_epoch = 50
learning_rate = 0.0001
print_freq = 1
batch_size = 128

train_params = network.all_params
train_op = tf.train.AdamOptimizer(learning_rate, beta1=0.9, beta2=0.999,
    epsilon=1e-08, use_locking=False).minimize(cost, var_list=train_params)

tl.layers.initialize_global_variables(sess)

network.print_layers()
for epoch in range(n_epoch):
    start_time = time.time()
    for X_train_a, y_train_a in tl.iterate.minibatches(
                                X_train, y_train, batch_size, shuffle=True):
         sess.run(train_op, feed_dict={x: X_train_a, y_: y_train_a})
    if epoch + 1 == 1 or (epoch + 1) % print_freq == 0:
        print("Epoch %d of %d took %fs" % (epoch + 1, n_epoch, time.time() - start_time))
        train_loss, train_acc, n_batch = 0, 0, 0
        for X_train_a, y_train_a in tl.iterate.minibatches(
                                 X_train, y_train, batch_size, shuffle=True):
            err, ac = sess.run([cost_test, acc], feed_dict={x: X_train_a, y_: y_train_a})
            train_loss += err; train_acc += ac; n_batch += 1
        print("   train loss: %f" % (train_loss/ n_batch))
        print("   train acc: %f" % (train_acc/ n_batch))
        test_loss, test_acc, n_batch = 0, 0, 0
        for X_test_a, y_test_a in tl.iterate.minibatches(
                                    X_test, y_test, batch_size, shuffle=True):
            err, ac = sess.run([cost_test, acc], feed_dict={x: X_test_a, y_: y_test_a})
            test_loss += err; test_acc += ac; n_batch += 1
        print("   test loss: %f" % (test_loss/ n_batch))
        print("   test acc: %f" % (test_acc/ n_batch))
sess.close()    
```

**Comment by zsdonghao** [comment]
you need to change your code from:
```python
network, cost, _ = model(x, y_,is_train=True,reuse=True)
_, cost_test, acc = model(x, y_, is_train=False,reuse=True)
```
to:
```python
network, cost, _ = model(x, y_,is_train=True,reuse=False)
_, cost_test, acc = model(x, y_, is_train=False,reuse=True)
```

and from:
```python
    with tf.variable_scope("model1", reuse=reuse):
        network = tl.layers.InputLayer(x,name='input')
```
to:
```python
    with tf.variable_scope("model1", reuse=reuse):
        tl.layers.set_name_reuse(reuse)
        network = tl.layers.InputLayer(x,name='input')
```
      

please check: https://github.com/wagamamaz/tensorlayer-tricks

**Comment by Nandayang** [comment]
thanks for your concern.it seems that i can only run this for a single time ,in the next time i run the code,it get the same error.if i close my spyder and reopen it ,the code can be run. is it a bug?@zsdonghao

**Comment by zsdonghao** [comment]
there is not a bug, you need to reuse your parameters if you want to use a model twice. please check the link i send you.

**Comment by Nandayang** [comment]
i m really sorry to disturb you,i just copy the file 'tutorial_cifar10.py' in your examples,the first time i run the code everything is ok,while the second gives an error just like i said.In the second time i just change the learning rate ,so if there is nothing wrong,does it means i have to reopen my spdyer another time if i  just want to change the params.It is so troublesome. <br>
Then i find it worked  through changing the `with tf.variable_scope("mode2", reuse=reuse)`,from model1 to model2 .<br>
Just wonder anyother methods to solve this questions.thank you for responsing when i asked so stupid questions.I love tensorlayer. @zsdonghao

**Comment by zsdonghao** [comment]
don't worry.
if you got error in second run, this mean you have not clear everything (variables, layer names etc) before your second one.

to remove layer names please use `tl.layers.clear_layers_name()` before the second run, if you got some expections about variable exists , you also need to clear the tensorflow variables.

however, if people want to run a script twice, people usually close the python process and run again.

**Comment by Nandayang** [comment]
Thanks for helping.Really appreciate
